tests/manage/z_cluster/nodes/test_node_replacement_proactive.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import re
+
+import pytest
+import random
+from ocs_ci.framework.testlib import (
+    tier4, ManageTest, aws_platform_required, ignore_leftovers, ipi_deployment_required)
+from ocs_ci.ocs import machine
+from ocs_ci.ocs import constants
+from ocs_ci.ocs import ocp
+from ocs_ci.ocs import node
+from tests.sanity_helpers import Sanity
+
+log = logging.getLogger(__name__)
+
+
+@tier4
+@ignore_leftovers
+@aws_platform_required
+@ipi_deployment_required
+class Testknip894(ManageTest):
+    """
+    Knip-678 Node Replacement
+    """
+
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+        """
+        self.sanity_helpers = Sanity()
+
+    def test_nodereplacement_proactive(self, pvc_factory, pod_factory):
+        """
+        Knip-894 Node Replacement proactive
+        """
+        self.sanity_helpers.create_resources(pvc_factory, pod_factory)
+        ocs_worker_node = node.get_ocs_worker_nodes()
+        machine_name = machine.get_machine_from_node_name(random.choice(ocs_worker_node))
+        log.info(f"Deleting machine {machine_name}")
+        machine.delete_machine_and_check_state_of_new_spinned_machine(machine_name)
+        new_machine_list = machine.get_machines()
+        for machines in new_machine_list:
+            if re.match(machines.name[:-6], machine_name):
+                new_machine_name = machines.name
+        machineset_name = machine.get_machineset_from_machine_name(new_machine_name)
+        log.info("Wating for new worker node to be in ready state")
+        machine.wait_for_new_node_to_be_ready(machineset_name)
+        new_node_obj = node.get_node_from_machine_name(new_machine_name)
+        log.info("Adding ocs label to newly created worker node")
+        node_obj = ocp.OCP(kind='node')
+        node_obj.add_label(
+            resource_name=new_node_obj.name,
+            label=constants.OPERATOR_NODE_LABEL
+        )
+        log.info(
+            f"Successfully labeled {new_node_obj.name} with OCS storage label"
+        )
+        self.sanity_helpers.health_check()
```

Comment:
this function appears to check mainly the ceph's health. 
Need to check als whether ocs pods are healthy or not.

Reply:
basically the exit criteria should consist of:
1) Data integrity checks
2) All OCS nodes status (all should be in Ready state)
3) Pods health check
4) ceph health check
5) to make sure the old node is not visible/active
6) include supported workloads like pgsql, logging, metrics, monitoring and check whether pvcs can be created and IOs can be done on them- (Note: this needs to be done in general for all the scripts)

@PrasadDesala please check above. it may apply to your tests also.....

Reply:
This function will check both health ceph as well as cluster health
https://github.com/red-hat-storage/ocs-ci/blob/master/tests/sanity_helpers.py#L27

Reply:
hope that includes all pods like cephfs and rbd plugin

Reply:
Can you confirm 1) 2) and 5) are taken care in the script? It would be great if you can implement 6 also.

Reply:
> hope that includes all pods like cephfs and rbd plugin

Yes it will check for all pods in openshift-storage namespace

Reply:
> Can you confirm 1) 2) and 5) are taken care in the script? It would be great if you can implement 6 also.
For 
1) if we use fio we can check data integrity, as well if pod is on same node which is going to get deleted pod will respin if it's dc if not dc it will get deleted, for dc after respin no io will be started automatically
2) will update in next commit
5) will update in next commit

Reply:
1) Please verify the data integrity for the pods on the surviving nodes. 
2) make use of dc pods only since this test involves node replacement. And as an exit criteria make sure the pods that moved to the new node from old node are able to re-run the IOs.

Reply:
Currently sanity helpers won't use DC pod if we merge this pr https://github.com/red-hat-storage/ocs-ci/pull/1331 sanity helpers will include DC pod